mm2src/coins/eth.rs

```diff
@@ -3185,6 +3354,44 @@ impl<T: TryToAddress> TryToAddress for Option<T> {
     }
 }
 
+pub trait GuiAuthMessages {
+    fn gui_auth_sign_message_hash(message: String) -> Option<[u8; 32]>;
+    fn generate_gui_auth_signed_validation(generator: GuiAuthValidationGenerator)
+        -> SignatureResult<GuiAuthValidation>;
+}
+
+impl GuiAuthMessages for EthCoin {
+    fn gui_auth_sign_message_hash(message: String) -> Option<[u8; 32]> {
+        let message_prefix = "atomicDEX Auth Ethereum Signed Message:\n";
+        let prefix_len = CompactInteger::from(message_prefix.len());
+
+        let mut stream = Stream::new();
+        prefix_len.serialize(&mut stream);
+        stream.append_slice(message_prefix.as_bytes());
+        stream.append_slice(message.len().to_string().as_bytes());
+        stream.append_slice(message.as_bytes());
+
+        Some(keccak256(&stream.out()).take())
+    }
+
+    fn generate_gui_auth_signed_validation(
+        generator: GuiAuthValidationGenerator,
+    ) -> SignatureResult<GuiAuthValidation> {
+        let timestamp_message = get_utc_timestamp() + 90; // 90 seconds to expire
+
+        let message_hash =
+            EthCoin::gui_auth_sign_message_hash(timestamp_message.to_string()).ok_or(SignatureError::PrefixNotFound)?;
+        let signature = sign(&generator.secret, &H256::from(message_hash))?;
+
+        Ok(GuiAuthValidation {
+            coin_ticker: String::from("ETH"),
```

Comment:
We already support a dozen of ETH forks (and QuickNode supports a few of them too), so `coin_ticker` should not be hard-coded. And it also should not be hard-coded in auth repo here https://github.com/KomodoPlatform/atomicdex-gui-auth/blob/main/main/src/security/proof_of_funding.rs#L22

Reply:
Will updated it.

For `atomicdex-gui-auth` the aim was supporting only ETH for the first release, so that's why it's hard coded there. It will be much different when we support more coins there.

Reply:
fixed via [ec33738](https://github.com/KomodoPlatform/atomicDEX-API/commit/ec3373863b79bd0c1324157abc436d2dd8ce0925) and [cde55a3](https://github.com/KomodoPlatform/atomicdex-gui-auth/commit/cde55a3c9f8771a096f78495d90e2bdac47bbb2d)